### ISSUE TITLE
Allow to use webjars rather than CDN in the web console

### ISSUE THREAD

**Issue by Dreampie** [restx/restx#129]
my  project  can't use  cdn，may I use  webjars？

```javascript
GET https://ajax.googleapis.com/ajax/libs/angularjs/1.0.8/angular.min.js net::ERR_CONNECTION_TIMED_OUT
GET https://ajax.googleapis.com/ajax/libs/angularjs/1.0.8/angular-resource.min.js net::ERR_CONNECTION_TIMED_OUT
GET https://ajax.googleapis.com/ajax/libs/angularjs/1.0.8/angular.min.js net::ERR_CONNECTION_TIMED_OUT
ui-bootstrap-tpls.min.js:1 Uncaught ReferenceError: angular is not defined
app.js:6 Uncaught ReferenceError: angular is not defined
admin.js:3 Uncaught ReferenceError: angular is not defined
Api.js:3 Uncaught TypeError: Cannot read property 'factory' of undefined
ApisController.js:3 Uncaught TypeError: Cannot read property 'controller' of undefined
OperationController.js:3 Uncaught TypeError: Cannot read property 'run' of undefined
EntityController.js:3 Uncaught TypeError: Cannot read property 'controller' of undefined

```

for  js lib like  this  in local,can't i config it or  use requirejs  for  cdn+local

```xml

<dependency>
    <groupId>org.webjars</groupId>
    <artifactId>angularjs</artifactId>
    <version>1.3.6</version>
</dependency>

```

**Comment by fcamblor** [comment]
Started to work on the topic, I have nice results.

Some drawbacks though :
- The webjar design, considering jar dependencies, doesn't allow to use same artefact in different versions at the same time
  This imply that if we put a dependency on `org.webjars:angularjs:1.0.5` in `restx-admin`, developers won't be able to use another version than this one without risking to bring incompatibilities on `restx-admin` module. Moreover, this doesn't allow to meld different versions of the same lib on different modules (today, we are using `angular` in versions `1.0.5`, `1.0.8`, `1.2.1` and `1.2.9` on the different restx modules).
- Moreover, some libraries used are not `webjar`-ed (for instance `jquery.event.drag`)
- We won't be able to reference webjars with absolute URIs (at least in `restx-admin` modules) because we're not knowing, clientside, restx api url root path (defaulted to `/api` but which can be changed)

To solve this issues, I'll create a dedicated restx webjar component containing every required js/css libraries used today. And maybe It would be an occasion to reduce the cluttering of versions used (using, at least, only `angular` `1.0.8` and `1.2.9`).


On the long term, I wonder if we shouldn't plug back some behaviour allowing to re-enable CDN (which would be better served on most of production environnments .. but not all that's why we should think it as a configuration option)
The best thing would be to provide a restx clientside library allowing to behave this ways depending on a serverside configuration option :
- Always use local webjars
- Only in dev mode : use CDN libraries and fallback to local libraries when offline is detected
- Always use CDN libraries
This would require to always provide CDN url + local urls when referencing webjars
... but I don't believe this feature is in the issue current scope and is required in the short term.


In terms of implementation, what I'm going to do (fill free to discuss this) :
- Provide a new `restx-webjars` module
- With a new `ResourcesRoute` binded on `/@/webjars/*` urls (that is to say, if you want to reference, saying, angular 1.0.8, you will have to GET `/<url-root-pattern>/@/webjars/angularjs/1.0.5/angular.min.js`)
- In restx admin modules, I'll reference webjars in a `relative path` manner (to not make asumptions on the `<url-root-pattern>`) instead of referencing CDNs


WDYT ?

**Comment by xhanin** [comment]
That sounds good, and I wonder if we will ever restore the option to use CDN: the admin console shouldn't be used so frequently, so the gain of using a CDN is not that interesting in that case IMO. 

What would be interesting to make the `ResourcesRoute` more efficient would be to add cache control to make the browser cache the resource, especially for these resources which should never change. But that's a different issue.

**Comment by fcamblor** [comment]
Merged on commit 986ba8cebba3b4d1bc63502311650f019d576c8b
